test/com/google/javascript/jscomp/runtime_tests/polyfill_tests/array_copywithin_test.js

```diff
@@ -64,5 +64,55 @@ testSuite({
     arr = {length: 3, 1: 4, 3: 'unused'};
     assertEquals(arr, Array.prototype.copyWithin.call(noCheck(arr), 0, 1));
     assertObjectEquals({length: 3, 0: 4, 3: 'unused'}, arr);
+
+    arr = {length: 3, 1: 4, 3: 'unused'};
+    assertEquals(arr, Array.prototype.copyWithin.call(noCheck(arr), 1, 0));
+    assertObjectEquals({length: 3, 2: 4, 3: 'unused'}, arr);
+  },
+
+  testCopyWithin_coercingArgs() {
+    assertObjectEquals([1, 2, 3, 3], [0, 1, 2, 3].copyWithin(NaN, 1));
+    assertObjectEquals([1, 2, 3, 3], [0, 1, 2, 3].copyWithin(0.5, 1));
+    assertObjectEquals([0, 0, 1, 2], [0, 1, 2, 3].copyWithin(1.5, 0));
+
+    assertObjectEquals([0, 0, 1, 2], [0, 1, 2, 3].copyWithin(1, NaN));
+    assertObjectEquals([0, 0, 1, 2], [0, 1, 2, 3].copyWithin(1, 0.5));
+    assertObjectEquals([1, 2, 3, 3], [0, 1, 2, 3].copyWithin(0, 1.5));
+
+    assertObjectEquals([0, 1, 2, 3], [0, 1, 2, 3].copyWithin(1, 0, NaN));
+    assertObjectEquals([0, 0, 2, 3], [0, 1, 2, 3].copyWithin(1, 0, 1.5));
+    assertObjectEquals([0, 0, 1, 3], [0, 1, 2, 3].copyWithin(1, 0, -2.5));
+    assertObjectEquals([0, 0, 1, 2], [0, 1, 2, 3].copyWithin(1, 0, undefined));
+  },
+
+  testCopyWithin_negativeArgs() {
+    assertObjectEquals([1, 2, 3, 1, 2], [1, 2, 3, 4, 5].copyWithin(-2, 0));
+    assertObjectEquals([1, 3, 4, 5, 5], [1, 2, 3, 4, 5].copyWithin(1, -3));
+    assertObjectEquals([1, 3, 4, 4, 5], [1, 2, 3, 4, 5].copyWithin(1, 2, -1));
+  },
+
+  testCopyWithin_throwsIfNullish() {
```

Comment:
This test no longer passes. Please remove it.

Reply:
Really? It should be pass and I believe it passes.

`this` is `null` and `this.length` is referred on the top of the polyfill function, so it should throw an error.
https://github.com/google/closure-compiler/pull/2902/files#diff-a5e0f0788d56ad46f3db1223fd3c8b5cR36

Reply:
I didn't look close enough: it fails because the `start` and `target` arguments are not optional (unfortunately, I don't think there's a way to run the polyfill tests externally, so there's no way for you to reproduce it). The best you can do is something like this:
 
`Array.prototype.copyWithin(null, some_other_array, 0);`

Reply:
@tjgq understood. It is not a runtime test, but a type check failed. I added the second and the third argument.